lib/logstash/filters/dns.rb

```diff
@@ -122,6 +122,12 @@ def resolve(event)
     @resolve.each do |field|
       is_array = false
       raw = event.get(field)
+
+      if raw.nil?
+        @logger.warn("DNS filter could not resolve missing field", :field => field)
```

Comment:
I think that's a good point. I think there are some interesting UX concerns here. Thinking through it a bit here are my thoughts:

I think there are two use cases for tagging multiple fields:

1. The user expects all N fields to always be there
2. The user expects some subset of N fields to always be there

Given 2, the user may not consider it a failure if at least one field is matched!

IMHO if you have more than 1 field then the expectations for tagging get complicated. We'd need a `tag_on_failure => tag_name` option *and* a `expect_all_fields => true|false` option for that.  At that point, I think we're getting very complex.

Since the failure in this case is due to null fields, I think its reasonable to expect that if the user does have that problem, that they could solve it with conditionals before or after this filter. If the failure was due to a bad lookup or something like that, that'd be different, but for a non-existent field I think its a different story.